BackendWebApp/src/main/java/uk/ac/ebi/biostd/webapp/application/rest/service/PendingSubmissionListFilter.java

```diff
@@ -0,0 +1,82 @@
+package uk.ac.ebi.biostd.webapp.application.rest.service;
+
+import uk.ac.ebi.biostd.webapp.application.rest.dto.PendingSubmissionListFiltersDto;
+import uk.ac.ebi.biostd.webapp.application.rest.dto.PendingSubmissionListItemDto;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Predicate;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+public class PendingSubmissionListFilter {
+
+    public static Predicate<? super PendingSubmissionListItemDto> asPredicate(PendingSubmissionListFiltersDto filters) {
+        List<Predicate<? super PendingSubmissionListItemDto>> predicates = new ArrayList<>();
+
+        Optional.ofNullable(filters.getAccNo())
```

Comment:
instead of this what you can do is add getter in  `public Optional<String> getaccNo()` PendingSubmissionListFiltersDto so is clear for everyone that property can be null and if we re use filter in other parts we can avoid a NPE

Reply:
Good point. I was not sure how it will work with lombok's @Data annotation

Reply:
Removed Data. Added Getter and Setter instead. Done. Looks better now. Thanks!